File: src/main/java/com/practice/web/dto/CategoryDto.java
package com.practice.web.dto;

import com.practice.theater.models.xml.Category;

import java.math.BigDecimal;

public class CategoryDto {
    private long id;
    private String name;
    private BigDecimal price;

    public static CategoryDto from(Category category) {
        CategoryDto dto = new CategoryDto();
        dto.id = category.getId();
        dto.name = category.getName();
        dto.price = category.getPrice();
        return dto;
    }

    public long getId() {
        return id;
    }

    public void setId(long id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public BigDecimal getPrice() {
        return price;
    }

    public void setPrice(BigDecimal price) {
        this.price = price;
    }
}

File: frontend/src/models/dto/CategoryDto.js
export class CategoryDto {

    /**
     * @param obj {{id: Number, name: String, price: Number}}
     */
    constructor(obj) {
        this._id = obj.id;
        this._name = obj.name;
        this._price = obj.price;
    }


    get id() {
        return this._id;
    }

    set id(value) {
        this._id = value;
    }

    get name() {
        return this._name;
    }

    set name(value) {
        this._name = value;
    }

    get price() {
        return this._price;
    }

    set price(value) {
        this._price = value;
    }
}